app/controllers/GatewayNPSController.scala

```diff
@@ -52,6 +55,7 @@ class GatewayNPSController @Inject() (
     val headers = httpResponse.headers
       .flatMap { case (key, values) => values.map(value => (key, value)) }
       .toSeq
+      .filter(pair => !notAllowCarryHeaders.contains(pair._1.toLowerCase))
```

Comment:
This maybe more efficient as it deals with all the headers in one step maybe?

private def mapHttpResponseToResult(httpResponse: HttpResponse): Result = {
  val status  = httpResponse.status
  val body    = httpResponse.body
  val headers = httpResponse.headers.toSeq
    .flatMap { case (key, values) =>
      val lowerKey = key.toLowerCase
      if (!notAllowCarryHeaders.contains(lowerKey)) values.map(value => (lowerKey, value))
      else Seq.empty
    }
    .sortBy(_._1)

  Status(status)(body).withHeaders(headers: _*)
}